### PR TITLE
Implement recovery features

### PR DESCRIPTION
**Pull request by pnobbe**
Includes updated postman collection

### Merge when:

* [ ] The merge build succeeded
* [ ] The PR has been reviewed and is approved
* [ ] All packages are upgraded to the latest version (only PR into master)

**What issue does this PR address?**

Implements the necessary recovery features for the upcoming client app feature upgrade.

**Does this PR introduce a breaking change?**

No

**Additional information**:

Remove branch after merge

### CODE REVIEW

**Review by yorickdewid — CHANGES_REQUESTED** [review]
* `IRecoverySampleRepository` warnings/errors need to addressed somehow. The build in `develop` must never report warnings/errors.
* About consistency; Even if some things are wrong or could be done better, abstain and submit as isolated patch. This makes reviews much easier and keeps a linear record of changes.

Postman file is ignored in the review

**Comment by yorickdewid** [comment]
Inquiry entities are subject to state control in which they can only move from one state (say 'pending') to another state ('pending_review'). This mechanism also applies to recoveries and is lacking. Account for this

**Comment by pnobbe** [comment]
Rebased from `develop`

File: src/FunderMaps.WebApi/Controllers/Report/RecoveryController.cs [hunk 1]
```diff
@@ -0,0 +1,375 @@
...
+        [HttpPost("{id:int}/status_review")]
+        [Authorize(Policy = "WriterAdministratorPolicy")]
+        public async Task<IActionResult> SetStatusReviewAsync(int id)
+        {
+            // Act.
+            Recovery recovery = await _recoveryRepository.GetByIdAsync(id);
+            Organization organization = await _organizationRepository.GetByIdAsync(_appContext.TenantId);
+            User creator = await _userRepository.GetByIdAsync(recovery.Attribution.Creator);
+            User reviewer = await _userRepository.GetByIdAsync(recovery.Attribution.Reviewer.Value);
+
```
Comment (yorickdewid) [inline src/FunderMaps.WebApi/Controllers/Report/RecoveryController.cs:221]: Remove empty line

File: src/FunderMaps.WebApi/Controllers/Report/RecoveryController.cs [hunk 2]
```diff
@@ -0,0 +1,375 @@
...
+            return NoContent();
+        }
+
+        // POST: api/recovery/{id}/status_approved
+        /// <summary>
+        ///     Set recovery status to done by id.
+        /// </summary>
+        [HttpPost("{id:int}/status_approved")]
+        [Authorize(Policy = "VerifierAdministratorPolicy")]
+
```
Comment (yorickdewid) [inline src/FunderMaps.WebApi/Controllers/Report/RecoveryController.cs:316]: Remove empty line

**Review by yorickdewid — CHANGES_REQUESTED** [review]

**Review by yorickdewid — APPROVED** [review]

**Pull request merged**